### ISSUE TITLE
Look into caching InternalConfigSource lookup results

### ISSUE THREAD

**Issue by jdmcclur** [OpenLiberty/open-liberty#19624]
Looking at profiles for some microprofile-primitives, I see some overhead on calls to getOptionalValue (coming from mpOpenTracing and mpJwt).
 
Some of the path is coming from here:
 ```
  Parent      0   0.02   2.10           1         101    J:io/openliberty/microprofile/config/internal/common/InternalConfigSource.getValue(Ljava/lang/String;)Ljava/lang/String;
 
    Self      0   0.02   2.10           1         101    J:io/openliberty/microprofile/config/internal/serverxml/ServerXMLVariableConfigSource.getProperties()Ljava/util/Map;
 
   Child      0   0.08   1.27           4          61    J:io/openliberty/microprofile/config/internal/serverxml/ServerXMLDefaultVariableConfigSource.getServerXMLVariables()Ljava/util/Map;
   Child      0   0.00   0.81           0          39    J:io/openliberty/microprofile/config/internal/serverxml/ServerXMLVariableConfigSource.getServerXMLVariables()Ljava/util/Map;
 ```
It would be great if we could cache the properties map in the two child methods above, and only regenerate it if something has changed, or if it's been 500 ms or something similar.

**Comment by Azquelt** [comment]
Profiling confirms that, as expected, most of the time is spent copying collections of variables which is done on every lookup.

Notably, the performance impact increases as more variables are added to the server.xml

In my profiling, running lookups for non-existent config properties in a tight loop, the ServerXML variable sources accounted for:
* 40% runtime with no variables in server.xml
* 67% runtime with 10 variables in server.xml
* 78% runtime with 10 variables and 10 variables declaring a default value in server.xml

**Comment by Azquelt** [comment]
The test fix in #19690 (which still needs review so may have further changes and probably won't be merged until the new year) improves the performance substantially.

With 10 variables and 10 variables declaring a default value in server.xml, the ServerXML variable sources dropped from 78% runtime to 4% runtime.

In addition, my benchmark runtime (30M lookups in a tight loop) dropped from 1m11s to 14s on my local machine.

The change stores additional maps which are updated whenever a variable changes rather than computing them on each lookup.
